lib/text_sender.rb

```diff
@@ -0,0 +1,24 @@
+require 'twilio-ruby'
+
+class TextSender
+
+  def initialize(time_class = Time, client_class = Twilio::REST::Client)
+    @time = time_class.new
+    @account_sid = ENV.fetch('TWILIO_ACCOUNT_SID')
+    @auth_token = ENV.fetch('TWILIO_AUTH_TOKEN')
+    @client = client_class.new(@account_sid, @auth_token)
+  end
+
+  def send_message
+    @client.messages.create(
+                           from: ENV.fetch('TWILIO_SEND_NUMBER'),
+                           body: generate_message_body,
```

Comment:
Does the twilio client give a response after the message has been created? A stretch task could be to use the response from Twilio to determine whether you need to retry.